### ISSUE TITLE
Make a simple mp-hp merged file, primarily for IMPC

### ISSUE THREAD

**Issue by cmungall** [obophenotype/upheno#162]
This should combine HP and MP, using both lexical and logically inferred relationships.

TBD: should equivalence cliques be merged (MP as primary), or left distinct, with equivalence axioms connecting?

**Comment by jmcmurry** [comment]
"Other phenotype" should be omitted from this merge, please

**Comment by cmungall** [comment]
Was attempt to use robot for the inference step, but blocked on this: https://github.com/ontodev/robot/issues/91

Switching to owltools for now

**Comment by cmungall** [comment]
Here is a screenshot of mp-hp-view.owl in Protege. Although the hierarchy is pre-inferred, it benefits from using the inferred hierarchy so equivalence pairs are grouped into Nodes:

<img width="1142" alt="screen shot 2016-07-22 at 8 23 46 am" src="https://cloud.githubusercontent.com/assets/50745/17061935/ad371bb0-4fe5-11e6-850d-72dbb7d1ea26.png">

**Comment by mellybelly** [comment]
I suspect that for ease of use, that equivalence cliques be merged with MP as primary

**Comment by cmungall** [comment]
Jenkins job: https://build.berkeleybop.org/job/build-mp-hp-view

Currently deployed here
http://build-artifacts.berkeleybop.org/build-mp-hp-view/latest/mp-hp-view.owl

Still to redirect purl here

**Comment by cmungall** [comment]
This should resolve shortly: http://purl.obolibrary.org/obo/upheno/mp-hp-view.owl

I configured the OBO purl server to point to the cloudfront distribution we use for the berkeley jenkins jobs. We can look to switching this to ci.mi for deployment later.

**Comment by cmungall** [comment]
currently there is no merge equivalent classes step:

https://github.com/obophenotype/upheno/blob/e12bd8a38f4dd7c85dcbd42fba3d6665bf3a71c9/Makefile#L154-L166

**Comment by nicolevasilevsky** [comment]
I am spot checked the file, generally looks good. 

One potential issue:
should bone marrow failure (MP_0013776) be equiv to decreased bone marrow cell number (MP_0000333)?

All of the phenotypes that users searched for in our user testing are in the new file, except alcohol withdrawal (which is neither in HP or MP). Does anyone know if that term is intentionally omitted from the phenotype ontologies or should I make a new term request?

**Comment by jmcmurry** [comment]
If @nicolevasilevsky says that the terms our users looked for are in there and structured appropriately, this sounds good to me. That's the best metric we have for whether it fulfills the use case.

**Comment by cmungall** [comment]
On 28 Jul 2016, at 13:24, nicolevasilevsky wrote:

> I am spot checked the file, generally looks good.
>
> One potential issue:
> should bone marrow failure (MP_0013776) be equiv to decreased bone 
> marrow cell number (MP_0000333)?

No - in fact we shouldn't have any intra-ontology equivalencies.

If you open mp-hp-view.owl in Protege (this is the source importer, has 
no inferences asserted) you can get an explanation for this

> All of the phenotypes that users searched for in our user testing are 
> in the new file, except alcohol withdrawal (which is neither in HP or 
> MP). Does anyone know if that term is intentionally omitted from the 
> phenotype ontologies or should I make a new term request?

It would be useful to know why the user was searching for this and what 
they expected. While it is certainly a phenotype in the sense of an 
observable property of an organism, it is not the kind of abnormal 
phenotype with genetic etiology we'd expect to see in HP or MP. Perhaps 
the users needs would be better served by a term describing addictive 
behavior or response to alcohol.

**Comment by nicolevasilevsky** [comment]
it doesn't give me an explanation, I just see this:
![image](https://cloud.githubusercontent.com/assets/6722114/17229209/80f26650-54cb-11e6-934a-4992f21cc8cb.png)

**Comment by jmcmurry** [comment]
The bone marrow inquiry was Maureen's (related to FA)

**Comment by cmungall** [comment]
@nicolevasilevsky can you look into these?

 * https://github.com/obophenotype/upheno/blob/master/reports/mp-mp-equivalent-classes.tsv
 * https://github.com/obophenotype/upheno/blob/master/reports/hp-hp-equivalent-classes.tsv

these either come from:

 1. false positive NER mapping
 2. bad logical axiom or combination of axioms
    - bad logical def equivalence axiom
    - logical defs are fine but subclass axioms between two ontologies disagree

**Comment by cmungall** [comment]
@nicolevasilevsky - the first thing to check is the axiom annotation - that should tell you if it comes from a textmatch

**Comment by nicolevasilevsky** [comment]
@cmungall yes, will do!

**Comment by nicolevasilevsky** [comment]
and the alcohol withdrawal search was from a user who used to study alcoholism in rodents, and I agree, there are other ontology terms that would probably suit her needs

**Comment by cmungall** [comment]
@nicolevasilevsky - sorry, you should be using mp-hp.owl. mp-hp-view.owl is the end-product file with inferences already performed

**Comment by nicolevasilevsky** [comment]
it is in MonDO-Reports

**Comment by cmungall** [comment]
It's in this repo: https://github.com/obophenotype/upheno/blob/master/mp-hp.owl

it's an importer

**Comment by nicolevasilevsky** [comment]
@cmungall I am looking at the mp-mp-equivalent-classes file and it doesn't look like there are bad logical defs. This example is super wrong, for abnormal thymus morphology (MP_0000703)

![image](https://cloud.githubusercontent.com/assets/6722114/17229986/e57de8a8-54ce-11e6-9e75-a14f08a6fe12.png)

It is making a lot of subclasses equiv to their parent.

**Comment by cmungall** [comment]
Right, the goal is to use the file without assertions to find explanations

**Comment by mellybelly** [comment]
One more task related to this ticket is to add terms to a slim based on the attached file
[mps.txt](https://github.com/obophenotype/upheno/files/391577/mps.txt)
These are the terms that are used for annotation and which will have pages created in IMPC, updating task list above. With apologies was lost in email.
